README.md

```diff
@@ -579,6 +580,314 @@ Note that the subflows are an array of repeating entries - one for each iteratio
+| DatabaseField                  | For fields from database columns. Represents a mapping between database fields such as `submittedAt`, `submissionId`, etc and their values. Does not include an iteration.                                              | Params: `String` database column name, `String` database field value <br><br> Examples:<br> new DatabaseField("submittedAt", "exampleSubmittedAtValue")                                                                                                                                                                                                |
+
+#### Creating a Custom Preparer
+
+When creating a custom preparer, you will want to create a new Java class that implements the
+`SubmissionFieldPreparer` interface and it's `prepareSubmissionFields` method.
+
+Depending on what you would like to do with your custom preparer, it should return a map of String
+input name to either `SingleField`, `CheckboxField`, `DatabaseField` or even a combination.
+
+For example, if you wanted to create a custom preparer that takes three seperate date fields and
+maps
+them to a single PDF
+
+```java
+public class ApplicantDateOfBirthPreparer implements SubmissionFieldPreparer {
+
+  @Override
+  public Map<String, SubmissionField> prepareSubmissionFields(Submission submission,
+      PdfMap pdfMap) {
+    Map<String, SubmissionField> submissionFields = new HashMap<>();
+
+    String month = submission.getInputData().get("applicantBirthMonth").toString();
+    String day = submission.getInputData().get("applicantBirthDay").toString();
+    String year = submission.getInputData().get("applicantBirthYear").toString();
+
+    submissionFields.put("applicantDateOfBirth",
+        new SingleField("applicantDateOfBirth", month + "/" + day + "/" + year, null));
+
+    return submissionFields;
+  }
+}
+```
+
+Note that in this example, we are using the `Submission` class to get the values of the inputs we
+want
+to map, and we are concatenating three separate fields into one and then creating a
+new `SingleField`
+object.
+
+#### Injecting custom input fields
+
+Note that in the above custom preparer example, we are creating a new `SingleField` for an input
+that
+does not actually exist in the application. Using custom preparers, we can create mappings for
+non-existent
+input fields. This is useful for creating mappings for fields that are not actually inputs in your
+application
+but are still needed for PDF generation.
+
+Note that you would need to create a field for this input in your `pdf-map.yaml` file. For example:
+
+```yaml
+inputFields:
+  applicantDateOfBirth: APPLICANT_DATE_OF_BIRTH
+```
+
+### Custom Preparers for Subflows
+
+Working with subflows is a bit different for custom preparers. In subflows, you will need to append
```

Comment:
I think we need to go into the "why" a little bit, explaining that we need to flatten the iteration data into something the PDF understands. Right now it's a list of maps in the inputData, but we need to map all that into the PDF field mapping, which has to be really specific as to where those fields map into. 

This should also talk about how we are expecting the PDF template file mappings to use the `_n` paradigm as well. We are making an assumption that the PDF file is using this pattern and we need to be clear to folks about that. 

Maybe some of that is best done in the  "Creating a template PDF file" section